Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.py

```diff
@@ -0,0 +1,1314 @@
+    sha256 = argToList(args.get('sha256'))
+    expiration_seconds = args.get('expiration_seconds')
+    download_to_xsoar = args.get('download_to_xsoar')
+
+    result = client.get_file_request(sha256, expiration_seconds)
+    contents = []
+
+    found_files = result.get('found', [])
+    for file_ in found_files:
+        contents.append({
+            'sha256': file_.get('sha256'),
+            'url': f"[{file_.get('url')}]({file_.get('url')})"
+        })
+
+        if download_to_xsoar == 'true':
+            request = requests.get(file_.get('url'))
+            demisto.results(fileResult('result.zip', request.content))
+
+    readable_output = tableToMarkdown('The file to download', contents)
+    results = CommandResults(
+        outputs_prefix='CarbonBlackEEDR.File',
+        outputs_key_field='sha256',
+        outputs=result,
+        readable_output=readable_output,
+        raw_response=result
+    )
+    return results
+
+
+def get_file_path_command(client: Client, args: Dict) -> CommandResults:
+    sha256 = args.get('sha256')
+
+    result = client.get_file_path_request(sha256)
+    readable_output = tableToMarkdown('The file path for the sha256', result)
+    results = CommandResults(
+        outputs_prefix='CarbonBlackEEDR.File',
+        outputs_key_field='sha256',
+        outputs=result,
+        readable_output=readable_output,
+        raw_response=result
+    )
+    return results
+
+
+def fetch_incidents(client: Client, fetch_time: str, fetch_limit: str, last_run: Dict) -> Tuple[List, Dict]:
+    last_fetched_alert_create_time = last_run.get('last_fetched_alert_create_time')
+    last_fetched_alert_id = last_run.get('last_fetched_alert_id', '')
+    if not last_fetched_alert_create_time:
+        last_fetched_alert_create_time, _ = parse_date_range(fetch_time, date_format='%Y-%m-%dT%H:%M:%S.000Z')
+    latest_alert_create_date = last_fetched_alert_create_time
+    latest_alert_id = last_fetched_alert_id
+
+    incidents = []
+
+    response = client.search_alerts_request(
+        sort_field='first_event_time',
+        sort_order='ASC',
+        create_time=assign_params(
+            start=last_fetched_alert_create_time,
+            end=datetime.now().strftime('%Y-%m-%dT%H:%M:%S.000Z')
+        ),
+        limit=fetch_limit,
+    )
+    alerts = response.get('results', [])
+
+    for alert in alerts:
+        alert_id = alert.get('id')
+        if alert_id == last_fetched_alert_id:
+            # got an alert we already fetched, skipping it
+            continue
+
+        alert_create_date = alert.get('create_time')
+        incident = {
+            'name': f'Carbon Black Enterprise EDR alert {alert_id}',
+            'occurred': alert_create_date,
+            'rawJSON': json.dumps(alert)
+        }
+        incidents.append(incident)
+        latest_alert_create_date = date_to_timestamp(datetime.strptime(alert_create_date, '%Y-%m-%dT%H:%M:%S.%fZ')) + 1
```

Comment:
are you sure you need to call `strptime` here? `date_to_timestamp` accepts both `datetime` objects and `str`

Reply:
yes. I tried the simple way, it's not a standard time stamp, when I tried with the command, it left me with .343z for. example If I remember correctly